### PR TITLE
330 - GroupElement - getEncoded

### PR DESCRIPTION
**Pull request by ApexTheory**
Hi @greenhat, please review at your convenience.  

Please note, this branch is based off my [previous negate branch](https://github.com/ergoplatform/sigma-rust/pull/334) (as it requires some common code), so it should be merged afterwards.

### CODE REVIEW

File: ergotree-ir/src/types/sgroup_elem.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    );
+}
+
+lazy_static! {
+    static ref GET_ENCODED_METHOD_DESC: SMethodDesc = SMethodDesc {
+        method_id: GET_ENCODED_METHOD_ID,
+        name: "getEncoded",
+        tpe: SFunc::new(
+            vec![SType::SGroupElement],
+            SType::SColl(SType::STypeVar(STypeVar::ov()).into()),
```
Comment (greenhat) [inline ergotree-ir/src/types/sgroup_elem.rs:42]: The result type should be `Coll[Byte]`.

**Review by greenhat — CHANGES_REQUESTED** [review]
Looking great! Only one comment regarding result type.

File: ergotree-ir/src/types/sgroup_elem.rs (hunk 1 above)
Comment (greenhat) [reply ergotree-ir/src/types/sgroup_elem.rs:42]: `STypeVar` is used when we need to express a polymorphic type (think `Coll.map`, etc.).

Comment (ApexTheory) [reply ergotree-ir/src/types/sgroup_elem.rs]: Fixed.

**Review by greenhat — APPROVED** [review]
Great! Thanks!

**Pull request merged**